cpu/esp32/esp-ieee802154/esp_ieee802154_hal.h

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (C) 2025 Gunar Schorcht
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+#pragma once
+
+/**
+ * @defgroup    cpu_esp32_esp_ieee802154 ESP32x IEEE 802.15.4 driver
+ * @ingroup     drivers_netdev
+ * @{
+ *
+ * @file
+ * @brief       ESP32x IEEE 802.15.4 driver
+ *
+ * @author      Gunar Schorcht <gunar@schorcht.net>
+ */
+
+#include <stdbool.h>
+
+#include "net/ieee802154/radio.h"
+
+/**
+ * @brief   Setup ESP32x in order to be used with the IEEE 802.15.4 Radio HAL
+ *
+ * @param[in] dev  pointer to the HAL descriptor associated to the device.
+ */
+void esp_ieee802154_setup(ieee802154_dev_t *dev);
+
+/**
+ * @brief Initialize the ESP32x IEEE 802.15.4 module.
+ *
+ * @return 0 on success
+ * @return negative errno on error
```

Comment:
```suggestion
 * @retval 0 on success
 * @retval negative errno on error
```

It might be nice to write *which* errno it would return. However in the current implementation, it will always return 0, even on failure.

Is it planned to change that behavior? Perhaps it would make sense to adapt the documentation otherwise.

Reply:
Difficult indeed. Will `esp_ieee802154_init` only ever be called by `auto_init_esp_ieee80215` or could it be expected that it'll be called by a custom program as well?

In the former case I'd say we could omit the return value, in the latter case it would be nice to handle the return.

In both cases, the documentation should reflect the actual behavior.

Reply:
:thinking: Theoretically, if someone doesn't use `auto_init` it could be that `esp_ieee802154_init` is called by the application.

On the othe hand, it is implemented exactly in the same way as all other IEEE 802.15.4 drivers, e.g., `mrjf24j40`, `kw2xrf_radio` and `nrf802154`. All of then document 
```
 * @return                  0 on success
 * @return                  <0 on error
```
but simply return 0.